tanner/emulators/base.py

```diff
@@ -113,11 +114,18 @@ def set_injectable_page(session):
         else:
             detection = await self.handle_get(session, data)
         
-        if 'payload' in detection and type(detection['payload']) is dict:
-            injectable_page = self.set_injectable_page(session)
-            if injectable_page is None:
-                injectable_page = '/index.html'
-            detection['payload']['page'] = injectable_page
+        if 'payload' not in detection:
+            detection['type'] = 1
+        elif 'payload' in detection:
+            if 'status_code' not in detection:
+                detection['type'] = 2
+                if detection['payload']['page']:
+                    injectable_page = self.set_injectable_page(session)
+                    if injectable_page is None:
+                        injectable_page = '/index.html'
+                    detection['payload']['page'] = injectable_page
+            else:
+                detection['type'] = 3
```

Comment:
Do we need that because usually that will be present in logs where it was present? And even if get it then how will we use it, as suppose we return `500` error to the client then sending any other data won't be needed.